content/posts/2021-06-25-Treasury-Expenditure-Bug.md

```diff
@@ -0,0 +1,167 @@
+---
+title:			"Treasury Expenditure Policy Bug"
+date:			2021-06-25
+authors:
+-  matheusd
+tags:			[Decred, Decentralized Treasury, consensus]
+banner_image:		"/images/posts/fiat-is-rigged.png"
+---
+
+Payments from the new decentralized treasury infrastructure are currently blocked and will be delayed until a new consensus vote takes place to implement a change to the consensus rules.  Contractor payouts will continue via the legacy treasury until the changes are deployed and activated.
+
+_No funds from either the decentralized treasury or from the legacy treasury are at risk of being lost._
+
+The only practical consequences of this issue are some inconvenience for operators of the Contractors Management System (CMS), the need for a new consensus vote to happen, and some delay for the new treasury to effectively start working.  _All funds in the network (including, but not limited to the treasury) are safe_.
+
+The issue resides in an overzealous check for the maximum expenditure policy. This policy defines the maximum value that a set of treasury spend transactions that are to be included in a block may draw from the treasury.
+
+Due to the test treasury spend transaction mined in mainnet on [block 556,716](https://explorer.dcrdata.org/tx/7507bcc72bfde895065034e12e6d462f2360163cd0c879f0db35514f9456b2c1), the maximum amount that can now be spent from the decentralized treasury account for the next few months is approximately 0.15 DCR, which is too low to be able to fulfill all monthly invoices in the CMS system.
+
+While no development team enjoys finding issues in shipped software and doubly so in distributed consensus code, we need to highlight that Decred's governance model and the way the decentralized treasury functions were introduced worked as designed to prevent a bigger problem.
+
+  - The migration plan from the legacy treasury to the decentralized treasury was created to handle any cases such as this.
+  - The payments to contractors will still continue as usual, coming from the legacy treasury wallet.
+  - There are no risks of the network forking, given the corresponding on-chain vote already activated the new rules and all older versions of the software are not following the main chain anymore.
+  - The decentralized treasury account is still accruing its 10% block reward.
+  - Finally, all the funds in the legacy treasury wallet are still intact and available for spending and eventually being moved to the new account system.
+
+
+The rest of this document will provide more technical details about the issue, how it was introduced, how it will be fixed and its impact to the network.
```

Comment:
```suggestion
The rest of this document will provide more technical details about the issue, how it was introduced, how it will be fixed, and its impact on the network.
```